packages/components/ng-ovh-payment-method/src/components/register/controller.js

```diff
@@ -82,104 +91,117 @@ export default class OvhPaymentMethodRegisterCtrl {
-              set(
-                paymentType,
-                'icon.data',
-                get(FALLBACK_IMAGES, paymentType.paymentType),
+    const promises = [
+      this.ovhPaymentMethodHelper.hasSpecificCrossBorderSentenceForCardPayment(),
+      this.ovhPaymentMethod.getAllAvailablePaymentMethodTypes(),
+    ];
+
+    return this.$q
+      .all(promises)
+      .then(
+        ([
+          hasSpecificCrossBorderSentenceForCardPayment,
+          paymentMethodsTypes,
+        ]) => {
+          // display or not the specific cross border sentence for given subsidiaries
+          this.hasSpecificCrossBorderSentenceForCardPayment = hasSpecificCrossBorderSentenceForCardPayment;
+          // sort available payment method types with the paymentMethodTypesOrder binding
+          this.availablePaymentMethodTypes.list = paymentMethodsTypes.sort(
+            (typeA, typeB) => {
+              const typeAIndex = this.paymentMethodTypesOrder.indexOf(
+                typeA.paymentType,
               );
-            }
-            return paymentType;
-          },
-        );
-
-        // split available payment method types list by chunk of paymentMethodTypesPerLine number
-        this.availablePaymentMethodTypes.chunks = chunk(
-          this.availablePaymentMethodTypes.list,
-          this.paymentMethodTypesPerLine,
-        );
-
-        const defaultPaymentMethodType = find(
-          this.availablePaymentMethodTypes.list,
-          {
-            paymentType: this.defaultPaymentMethodType,
-            ...(this.defaultPaymentMethodIntegration
-              ? { integration: this.defaultPaymentMethodIntegration }
-              : {}),
-          },
-        );
-
-        // set selected payment method type model
-        if (
-          !has(this.model, 'selectedPaymentMethodType') ||
-          isNil(this.model.selectedPaymentMethodType)
-        ) {
-          this.model.selectedPaymentMethodType = defaultPaymentMethodType;
-        } else if (this.model.selectedPaymentMethodType) {
-          // if the selected payment method type does not exist
-          // set the default one
-          const isModelTypeExists = some(
+              const typeBIndex = this.paymentMethodTypesOrder.indexOf(
+                typeB.paymentType,
+              );
+              return typeAIndex > typeBIndex;
+            },
+          );
+
+          // add a fallback image in case of no image provided by API (for legacies payment methods)
+          this.availablePaymentMethodTypes.list = map(
+            this.availablePaymentMethodTypes.list,
+            (paymentTypeParam) => {
+              const paymentType = paymentTypeParam;
+              if (!get(paymentType, 'icon.data')) {
+                set(
+                  paymentType,
+                  'icon.data',
+                  get(FALLBACK_IMAGES, paymentType.paymentType),
+                );
+              }
+              return paymentType;
+            },
+          );
```

Comment:
```suggestion
          this.availablePaymentMethodTypes.list = this.availablePaymentMethodTypes.list.map(
            (paymentTypeParam) => {
              const icon = paymentTypeParam.icon?.data;
              return { ...paymentTypeParam, icon: { data: icon || FALLBACK_IMAGES[paymentType] }};
            },
          );
```

Reply:
@ghyenne : this part of code was already existing, it doesn't concern my changes. I can modify if you want but your suggestion is not really optimized. Indeed, each object will be duplicated even if icon property is already set. Are you sure about that ?